packages/effect/test/Stream/do-notation.test.ts

```diff
@@ -37,6 +47,21 @@ describe("do notation", () => {
       pipe(Stream.fail("left"), Stream.bindTo("a"), Stream.bind("b", () => Stream.succeed(2))),
       "left"
     )
+    expectRight(
+      pipe(
+        Stream.succeed(1),
+        Stream.bindTo("a"),
+        (x) =>
+          pipe(
+            x,
+            // @ts-expect-error
+            Stream.bind("__proto__", ({ a }) => Stream.succeed(a + 1))
+          ) as Stream.Stream<{ a: number; __proto__: number }, unknown, never>,
```

Comment:
At present, if the name param to Stream.bind is inferred as never, the reader type infers as unknown instead of never, which breaks the call to expectRight.

This is already true if you attempt to shadow a name using Stream.bind today, illustrated here: https://effect.website/play/#71d73fbd5124